### PR TITLE
Adding CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Djelibeybi**
Provide documented contribution rules and guidelines to assist new users on how best to structure their Dockerfile or updates for the quickest review/merge process.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+submit your pull request.
+
+If you have any questions about a possible submission, feel free to open
+an issue too.
+
+## Pull Request Process
+1. Fork this repository
+2. Create a branch in your fork to implement the changes. We recommend using
+the issue number as part of your branch name, e.g. `1234-fixes`
+3. Ensure that any documentation is updated with the changes that are required
```
Comment (gvenzl) [inline CONTRIBUTING.md]: There should be another point saying something like "Ensure that samples are updated if the PR causes the behavior of the sample to change"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+1. Specify a fixed version in the `FROM` directive, i.e. use
+`FROM oraclelinux:7-slim` or `FROM java/serverjre:8`.
+
+### Security-related Rules
+
+1. Do not require the use of the `--privileged` flag when running a container.
+
+1. Do not run `ssh` inside a container.
+1. Do not use host networking mode (`--net=host`) for a container.
+1. Do not hard-code any passwords.
```
Comment (gvenzl) [inline CONTRIBUTING.md]: This should probably elaborate on that it's best practice to auto generate a password by default via openssl rand and accept a password as argument.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+The following are some guidelines that will not prevent an image from being
+merged, but are generally frowned upon if breached.
+
+- Don't install all possible required RPMs, even if the product
+documentation says so. Some RPMs aren't applicable inside a container, e.g
+filesystem utilities (`btrfs-progs`, `ocfs2-tools`, `nfs-utils`).
+- Don't install any interactive/user tools, e.g. things like `vim`, `less` or
+`man`. Debugging should be done prior to the image submission.
+- Don't install `wget` as the base images already include `curl`.
+- Always remember to run `yum clean all` in the same `RUN` directive as a
```
Comment (gvenzl) [inline CONTRIBUTING.md]: Include that the image should in general be kept as small as possible and unnecessary packages/parts removed.

**Comment by Djelibeybi** [comment]
I've pushed some changes. Great feedback, thanks @gvenzl. Can you review again, please?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# Contributing
+Oracle welcomes contributions to this repository from anyone.
+
+If you want to submit a pull request to fix an bug or enhance an existing
```
Comment (brunoborges) [inline CONTRIBUTING.md]: "fix a bug".

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+1. Submit the pull request. *Do not leave the pull request blank*. Explain exactly
+what your changes are meant to do and provide simple steps on how to validate
+your changes. We will assign the pull request to 2-3 people for review before
+it is merged.
+
+## Golden Rules
+
+We have some golden rules that we require all submitted `Dockerfiles` to abide
+by. These rules are provided by Oracle Global Product Security and may change
+at any time.
```
Comment (brunoborges) [inline CONTRIBUTING.md]: These rules are of course applied to all Oracle employees submitting contributions, but they will be _checked_ by us when contributions are coming from 3rd-parties. Maybe the wording here should be more directed to employees?

Comment (Djelibeybi) [inline CONTRIBUTING.md]: No, these rules apply to both employees and non-employees alike. We will not accept any PRs that don't follow these rules.

**Comment by nirmay** [comment]
looks good to me. Thanks for adding the rules.

**Comment by Djelibeybi** [comment]
Excellent feedback, @mriccell.  I've added that note and added some text around documenting any defaults that are used in the case that custom values are not provided.

**Comment by Djelibeybi** [comment]
Thanks team. Merged. If you have any other guidelines/rules, feel free to open a PR to include those.

**Pull request merged**